openraft/src/entry.rs

```diff
@@ -0,0 +1,156 @@
+use std::fmt::Debug;
+
+use crate::LogId;
+use crate::Membership;
+use crate::MessageSummary;
+use crate::NodeId;
+use crate::RaftTypeConfig;
+
+/// Defines operations on an entry payload.
+pub trait RaftPayload<NID: NodeId> {
+    /// Return `Some(())` if the entry payload is blank.
+    fn get_blank(&self) -> Option<()>;
+
+    /// Return `Some(&Membership)` if the entry payload is a membership payload.
+    fn get_membership(&self) -> Option<&Membership<NID>>;
+}
+
+/// Defines operations on an entry.
+pub trait RaftEntry<NID: NodeId>: RaftPayload<NID> {
```

Comment:
seems `RaftPayload` and `RaftEntry` has only definition and cannot see any using?

Reply:
It will be used in the next PR: to extract raft protocol out of `RaftCore`.

The protocol engine must not depend on `AppData`, or `AppDataResponse`.
These two traits provide APIs for it to access Entry.